sdk/keyvault/azcertificates/pollers.go

```diff
@@ -0,0 +1,136 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package azcertificates
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/keyvault/azcertificates/internal/generated"
+)
+
+type beginCreateCertificateOperation struct {
+	PollURL string
+	Status  string
+	poll    func(context.Context, string) (*http.Response, error)
+	result  func(context.Context) (CreateCertificateResponse, error)
+}
+
+func (b *beginCreateCertificateOperation) Done() bool {
+	return b.Status == "completed" || b.Status == "cancelled"
+}
+
+func (b *beginCreateCertificateOperation) Poll(ctx context.Context) (*http.Response, error) {
+	resp, err := b.poll(ctx, b.PollURL)
+	if err != nil {
+		return nil, err
+	}
+	if !runtime.HasStatusCode(resp, http.StatusOK, http.StatusAccepted) {
+		return nil, runtime.NewResponseError(resp)
+	}
+	payload, err := runtime.Payload(resp)
+	if err != nil {
+		return nil, err
+	}
+
+	var op Operation
+	if err := json.Unmarshal(payload, &op); err != nil {
+		return nil, err
+	}
+
+	if op.Status == nil {
+		return nil, errors.New("missing status")
+	}
+	b.Status = *op.Status
+	return resp, nil
+}
+
+func (b *beginCreateCertificateOperation) Result(ctx context.Context, out *CreateCertificateResponse) error {
+	result, err := b.result(ctx)
+	if err != nil {
+		return err
+	}
+	*out = result
+	return nil
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+type beginDeleteCertificateOperation struct {
+	resp *http.Response
+	poll func(context.Context) (*http.Response, error)
+}
+
+func (b *beginDeleteCertificateOperation) Done() bool {
+	return b.resp != nil && b.resp.StatusCode == http.StatusOK
```

Comment:
The client may be allowed to delete certs but forbidden to get them, in which case every poll will get a 403
```suggestion
	return runtime.HasStatusCode(b.resp, http.StatusOK, http.StatusForbidden)
```

Reply:
This makes me think that instead we should remove 403 as successful status code in the `Poll()` methods so that an error is returned.  Thoughts?

Reply:
The 403s have been removed from the list of successful status codes.